File: java/src/main/java/recipes/PriceSnapshot.java
// Compare one symbol's price across several exchanges.
package recipes;

import io.github.ccxt.exchanges.Binance;
import io.github.ccxt.exchanges.Kraken;
import io.github.ccxt.exchanges.Okx;
import io.github.ccxt.types.Ticker;

public class PriceSnapshot {

    public static void main(String[] args) {
        String symbol = args.length > 0 ? args[0] : "BTC/USDT";
        System.out.println(symbol);
        print("binance", new Binance().fetchTicker(symbol));
        print("kraken", new Kraken().fetchTicker(symbol));
        print("okx", new Okx().fetchTicker(symbol));
    }

    static void print(String id, Ticker ticker) {
        String spread = (ticker.bid != null && ticker.ask != null)
            ? String.format("%.4f%%", (ticker.ask - ticker.bid) / ticker.ask * 100)
            : "n/a";
        System.out.printf("%-8s last %s  bid %s  ask %s  spread %s%n", id, ticker.last, ticker.bid, ticker.ask, spread);
    }
}

File: csharp/recipes/PriceSnapshot.cs
// Compare one symbol's price across several exchanges.
using ccxt;

namespace Recipes;

public static class PriceSnapshot
{
    public static async Task Run(string[] args)
    {
        var symbol = args.Length > 0 ? args[0] : "BTC/USDT";
        var exchanges = new List<Exchange> { new Binance(), new Kraken(), new Okx() };
        Console.WriteLine(symbol);
        foreach (var exchange in exchanges)
        {
            var ticker = await exchange.FetchTicker(symbol);
            var spread = (ticker.bid != null && ticker.ask != null)
                ? $"{(ticker.ask - ticker.bid) / ticker.ask * 100:F4}%"
                : "n/a";
            Console.WriteLine($"{exchange.id,-8} last {ticker.last}  bid {ticker.bid}  ask {ticker.ask}  spread {spread}");
        }
    }
}
